### PR TITLE
Modifying 2-benchmark.js.

### PR DESCRIPTION
**Pull request by ALEXKIRNAS**
Adding new argument - array of functions names. Name choose in next order:
1. From array of names.
2. From function attribute `fn.name`.
3. Default name - 'Anonymius'.

### CODE REVIEW

File: JavaScript/2-benchmark.js [hunk 1]
```diff
@@ -20,7 +21,11 @@ benchmark.do = (count, retry, tests) => {
     }
     const end = process.hrtime(begin);
     const diff = end[0] * 1e9 + end[1];
-    const prefix = '.'.repeat(35 - (diff + fn.name).length);
-    console.log(fn.name + prefix + diff + ' nanoseconds');
+
+    const fromArray = namesOfTests && namesOfTests[orderNumber];
+    const fnName = fromArray || fn.name || 'Anonymius';
```
Comment (aqrln) [inline JavaScript/2-benchmark.js]: Do you mean "Anonymous"?

File: JavaScript/2-benchmark.js [hunk 2]
```diff
@@ -20,7 +21,11 @@ benchmark.do = (count, retry, tests) => {
...
+    const prefix = '.'.repeat(35 - (diff + fnName).length);
+
+    console.log(fnName + prefix + diff + ' nanoseconds');
   }
-};
+};
```
Comment (aqrln) [inline JavaScript/2-benchmark.js]: Ugh... Newline, again.

File: JavaScript/2-benchmark.js [hunk 3]
```diff
@@ -3,15 +3,16 @@
 const benchmark = {};
 module.exports = benchmark;
 
-benchmark.do = (count, retry, tests) => {
+benchmark.do = (count, retry, tests, namesOfTests) => {
```
Comment (aqrln) [inline JavaScript/2-benchmark.js]: I'd find it more readable to have this variable named `testNames` or something like that.

File: JavaScript/2-benchmark.js [hunk 4]
```diff
@@ -3,15 +3,16 @@
...
 
   let k;
   for (k = 0; k < retry; k++) {
-    tests.map(test);
+    let counter = 0; // Order number of function that will be call
```
Comment (aqrln) [inline JavaScript/2-benchmark.js]: I prefer single-line comments to be separated from code with two spaces, but I believe it is not required by our code style, so you may ignore this :)

Comment (aqrln) [inline JavaScript/2-benchmark.js]: Aside from that, consider rewording the comment itself. First of all, please fix a grammar mistake (s/call/called). Secondly, "order number" is quite a harsh word-to-word translation from Russian. In English it sounds like someone has ordered to buy a function and this variable contains the ID of their order :) "Index" would be fine here.

File: JavaScript/2-benchmark.js [hunk 5]
```diff
@@ -3,15 +3,16 @@
...
+    tests.map((fn) => test(fn, counter++));
     console.log();
   }
 
-  function test(fn) {
+  function test(fn, orderNumber) {
```
Comment (aqrln) [inline JavaScript/2-benchmark.js]: Ditto. Consider `count` or `index`.

File: JavaScript/2-benchmark.js [hunk 6]
```diff
@@ -20,7 +21,11 @@ benchmark.do = (count, retry, tests) => {
     }
     const end = process.hrtime(begin);
     const diff = end[0] * 1e9 + end[1];
-    const prefix = '.'.repeat(35 - (diff + fn.name).length);
-    console.log(fn.name + prefix + diff + ' nanoseconds');
+
+    const fromArray = namesOfTests && namesOfTests[orderNumber];
```
Comment (aqrln) [inline JavaScript/2-benchmark.js]: `fromArray` isn't very obvious. Maybe `explicitName` or something like that?

**Comment by ALEXKIRNAS** [comment]
@aqrln  Fixed.

**Comment by aqrln** [comment]
LGTM if it works with the current benchmarks.

**Comment by ALEXKIRNAS** [comment]
@aqrln What do you mean?

**Pull request closed without merging**